src/dawn/engine_dawn.cc

```diff
@@ -334,6 +335,78 @@ Result EngineDawn::DoClearDepth(const ClearDepthCommand* command) {
   return {};
 }
 
+uint32_t Align(uint32_t value, size_t alignment) {
+  assert(alignment <= UINT32_MAX);
+  assert(alignment != 0);
+  uint32_t alignment32 = static_cast<uint32_t>(alignment);
+  return (value + (alignment32 - 1)) & ~(alignment32 - 1);
+}
+
+MapResult MapTextureToHostBuffer(const RenderPipelineInfo& render_pipeline,
+                                 const ::dawn::Device& device) {
+  const auto width = render_pipeline.pipeline->GetFramebufferWidth();
+  const auto height = render_pipeline.pipeline->GetFramebufferHeight();
+  const auto pixelSize = render_pipeline.pipeline->GetColorAttachments()[0]
+                             .buffer->GetFormat()
+                             ->SizeInBytes();
```

Comment:
Once #466 lands this can just be buffer->GetTexelStride(), but has to be this for now.